pyopencl/__init__.py

```diff
@@ -1970,8 +1970,10 @@ def enqueue_copy(queue, dest, src, **kwargs):
 
         # These are NOT documented. They only support consistency with the
         # Buffer-based API for the sake of the Array.
-        assert kwargs.pop("src_offset", 0) == 0
-        assert kwargs.pop("dst_offset", 0) == 0
+        if kwargs.pop("src_offset", 0) != 0:
+            raise AssertionError
+        if kwargs.pop("dst_offset", 0) != 0:
+            raise AssertionError
```

Comment:
It'd be better if (all of) these were `ValueError`s with an error message.

Reply:
What do you think of fd856a5?

Reply:
That works, thanks!